test/system/helpers.bash

```diff
@@ -255,38 +201,63 @@ function defer-assertion-failures() {
 # Basic teardown: remove all pods and containers
 function basic_teardown() {
     echo "# [teardown]" >&2
-    local actions=(
-        "pod rm -t 0 --all --force --ignore"
-            "rm -t 0 --all --force --ignore"
-        "network prune --force"
-        "volume rm -a -f"
-    )
-    for action in "${actions[@]}"; do
-        run_podman '?' $action
 
-        # The -f commands should never exit nonzero, but if they do we want
-        # to know about it.
-        #   FIXME: someday: also test for [[ -n "$output" ]] - can't do this
-        #   yet because too many tests don't clean up their containers
-        if [[ $status -ne 0 ]]; then
-            echo "# [teardown] $_LOG_PROMPT podman $action" >&3
-            for line in "${lines[*]}"; do
-                echo "# $line" >&3
+    # Unlike normal tests teardown will not exit on first command failure
+    # but rather only uses the return code of the teardown function.
+    local exit_code=0
+    # Good tests will cleanup themselves so don't waste time
+    # running serval podman commands.
+    if [[ "$BATS_TEST_COMPLETED" -eq 1 ]]; then
+        # To speed CI up do not do this by default and only provide this as opt in option.
+        if [ -n "$PODMAN_BATS_LEAK_CHECK" ]; then
```

Comment:
These two `if`s can be combined into one, if you nuke lines 249-253 and add a conditional (on `$exit_code`) to line 256. (Ugh. Except that now `$exit_code` is overloaded. I'll leave that as an exercise for the reader).